addons/google_calendar/models/google_sync.py

```diff
@@ -166,20 +166,67 @@ def _sync_google2odoo(self, google_events: GoogleEvent, default_reminders=()):
 
         return synced_records
 
+    def _google_error_handling(self, result):
+        # We only handle the most problematic errors of sync events.
+        if result.response.status_code in (403, 400):
+            response = json.loads(result.response.text)
+            message = response['error'].get('message')
+            if self._name == 'calendar.event':
+                start = self.start.strftime('%Y-%m-%d at %H:%M')
+                event_ids = self.id
+                name = self.name
+                error_msg = "Error while syncing event: "
+                event = self
+            else:
+                # calendar recurrence is triggering the error
+                start = self.base_event_id.start.strftime('%Y-%m-%d at %H:%M')
+                event_ids = _("%(id)s and %(length)s following", id=self.base_event_id.id, length=len(self.calendar_event_ids.ids))
+                name = self.base_event_id.name
+                # prevent to sync other events
+                self.calendar_event_ids.need_sync = False
+                error_msg = "Error while syncing recurrence: "
+                event = self.base_event_id
+
+            # We don't have right access on the event or the request paramaters were bad.
+            if result.response.status_code == 400 and "invalid attendee email" in message.lower():
+                reason = "invalid attendee email. Please, verify the partners emails."
+            elif result.response.status_code == 403 and "'forbiddenForNonOrganizer'" in result.response.text:
```

Comment:
Is this error somewhere precise in the json, like `response['error']['code']` or something?

Reply:
When we get a 400, the error is generic. We can have multiple reasons for a 400 (invalid email, invalid recurrence rule etc). The json looks like this:
```
{'error': {'errors': [{'domain': 'global', 'reason': 'invalid', 'message': 'Invalid attendee email.'}], 'code': 400, 'message': 'Invalid attendee email.'}}
```
For an invalid recurrent rule for example, we get this error:
```
{'error': {'errors': [{'domain': 'global', 'reason': 'invalid', 'message': 'Invalid recurrence rule.'}], 'code': 400, 'message': 'Invalid recurrence rule.'}}
```

As @tfr-odoo found that there are a lot of errors triggered by these invalid emails that the user should fix. That's why we decided to notify the user in that case. And that explains why I have to read the message part and not the generic reason.

In case of a 403, the reason is more explicit:
```
[{'domain': 'calendar', 'reason': 'forbiddenForNonOrganizer', 'message': 'Shared properties can only be changed by the organizer of the event.'}]
```